Source/Core/Core/State.cpp

```diff
@@ -578,62 +745,50 @@ static void LoadFileStateData(const std::string& filename, std::vector<u8>& ret_
   }
 
   StateHeader header;
-  if (!f.ReadArray(&header, 1))
-  {
-    Core::DisplayMessage("State not found", 2000);
+  if (!ReadStateHeaderFromFile(header, f) || !ValidateHeaders(header))
     return;
-  }
 
-  if (strncmp(SConfig::GetInstance().GetGameID().c_str(), header.gameID, 6))
+  StateExtendedHeader extended_header;
+  f.ReadArray(&extended_header.base_header, 1);
+  // If StateExtendedHeader is ammended to include more than the base, add ReadBytes() calls here.
+  // Do NOT use ReadArray, as this would prevent future revisions from reading in char arrays.
+
+  if (extended_header.base_header.header_version != EXTENDED_HEADER_VERSION)
   {
-    Core::DisplayMessage(fmt::format("State belongs to a different game (ID {})",
-                                     std::string_view{header.gameID, std::size(header.gameID)}),
-                         2000);
-    return;
+    const std::string message =
+        fmt::format("Savestate extended header mismatch!\n\n"
+                    "Current: {}\n"
+                    "Savestate: {}",
+                    EXTENDED_HEADER_VERSION, extended_header.base_header.header_version);
+    Core::DisplayMessage(message, OSD::Duration::NORMAL);
```

Comment:
This is a good log message but a poor user-facing string, for the user I'd just put 'state corrupted' or similar. Also I assume you want to return here?